src/elements/ExpiryIndicator.vue

```diff
@@ -0,0 +1,87 @@
+<script setup lang="ts">
+import ExpiryIcon from '@/icons/ExpiryIcon.vue';
+import {computed} from "vue";
+import { t } from '@/composable/i18n';
+import {ExpiryUnitTypes} from "@/definitions";
+
+// component properties
+interface Props {
+  timer?: number;
+  warn?: number;
+  unit?: ExpiryUnitTypes;
+}
+const props = withDefaults(defineProps<Props>(), {
+  timer: 10,
+  warn: 5,
+  unit: ExpiryUnitTypes.Days
+})
+
+// Changes label from:
+// EXPIRES IN (n) DAYS > 5 days
+// EXPIRES IN 1 DAYS == 1 day
+// EXPIRED <= 0 days
+const timer = computed(() => props.timer);
+const warn = computed(() => props.warn); //timers before using the warning color
```

Comment:
You can use https://vuejs.org/api/reactivity-utilities.html#toref or https://vuejs.org/api/reactivity-utilities.html#torefs to do this automagically!